syncer/syncer_test.go

```diff
@@ -130,16 +130,28 @@ func TestStartAndShutdown(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.TODO())
 	ts := newTestSyncer(ctx, t, time.Millisecond*5)
 
-	syncedCh := make(chan struct{})
-	ts.syncer.RegisterChForSynced(context.TODO(), syncedCh)
+	syncedCh := ts.syncer.RegisterForSynced(context.TODO())
+	atxSynchedCh := ts.syncer.RegisterForATXSynced()
 
 	require.False(t, ts.syncer.IsSynced(context.TODO()))
 	require.False(t, ts.syncer.ListenToATXGossip())
 	require.False(t, ts.syncer.ListenToGossip())
 
 	// the node is synced when current layer is <= 1
 	ts.syncer.Start(context.TODO())
-	<-syncedCh
+
+	select {
+	case <-syncedCh:
+	case <-time.After(1 * time.Second):
+		require.Fail(t, "node should be synced")
+	}
```

Comment:
A helper would be nice as this pattern is used **a lot**.
```suggestion
	requireClosed(t, syncedCh, time.Second, "node should be synced")
```

Reply:
I prefer being explicit over factoring out small functions that are only used in a few places.

The `select` doesn't check if the channel is closed, only if it blocks upon a read. Checking for it to be closed requires additional steps that aren't necessary for what's being asserted here. In some places the inverse is done, checking if it blocks.

Introducing two additional functions with trivial logic to only save 3 lines of code is imho not aiding readability.